src/client/sandbox/node/live-node-list/dom-mutation-tracker.js

```diff
@@ -0,0 +1,81 @@
+import nativeMethods from '../../native-methods';
+import { getTagName, isDocumentFragmentNode, isShadowUIElement } from '../../../utils/dom';
+
+const MAX_SAFE_INTEGER = Math.pow(2, 53) - 1;
+const MIN_SAFE_INTEGER = -MAX_SAFE_INTEGER;
+
+class DOMMutationTracker {
+    constructor () {
+        this._tags               = nativeMethods.objectCreate.call(window.Object, null);
+        this._isDomContentLoaded = false;
+
+        nativeMethods.addEventListener.call(document, 'DOMContentLoaded', () => {
+            this._isDomContentLoaded = true;
+        });
+    }
+
+    _updateVersion (tagName) {
+        if (tagName in this._tags) {
+            if (this._tags[tagName] === MAX_SAFE_INTEGER)
+                this._tags[tagName] = MIN_SAFE_INTEGER;
+            else
+                ++this._tags[tagName];
+        }
+        else
+            this._tags[tagName] = MIN_SAFE_INTEGER;
+    }
+
+    _processElement (el) {
+        if (!el.tagName || isShadowUIElement(el))
+            return;
+
+        const tagName = getTagName(el);
+
+        this._updateVersion('*');
+        this._updateVersion(tagName);
+    }
+
+    _processAllChildren (el) {
```

Comment:
`_processChildren`

Reply:
This function process all children. The `_processChildren` name tell me that function process only children of element.